### PR TITLE
[ADD] account_invoice_pricelist: use pricelist when calculating prices

### PR DESCRIPTION
**Pull request by hbrunn**
This adds support for adding invoice lines by hand while still getting the correct price from the chosen pricelist

### CODE REVIEW

File: account_invoice_pricelist/model/account_invoice_line.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        result = super(AccountInvoiceLine, self).product_id_change(
+            product, uom_id, qty=qty, name=name, type=type,
+            partner_id=partner_id, fposition_id=fposition_id,
+            price_unit=price_unit, currency_id=currency_id,
+            company_id=company_id)
+        values = result.get('value', {})
+        if product and 'pricelist_id' in self.env.context:
+            # get pricelist price
+            values['price_unit'] = self.env['product.product']\
+                .browse([product])\
```
Comment (pedrobaeza) [inline account_invoice_pricelist/model/account_invoice_line.py:40]: You don't need to enclose the id between parenthesis for having browse working.

File: account_invoice_pricelist/model/account_invoice_line.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                .with_context(
+                    quantity=qty, pricelist=self.env.context['pricelist_id'],
+                    partner=partner_id)\
+                ._product_price(None, None)\
+                .get(product)
+            # adjust for currency, uos (taken from super)
+            if company_id and currency_id:
+                company = self.env['res.company'].browse([company_id])
```
Comment (pedrobaeza) [inline account_invoice_pricelist/model/account_invoice_line.py:48]: The same for parenthesis

File: account_invoice_pricelist/model/account_invoice_line.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                product = self.env['product.product'].browse([product])
+                currency = self.env['res.currency'].browse([currency_id])
+                if company.currency_id != currency:
+                    if type in ('in_invoice', 'in_refund'):
+                        values['price_unit'] = product.standard_price
```
Comment (pedrobaeza) [inline account_invoice_pricelist/model/account_invoice_line.py:53]: Why are you forcing here the price_unit to standard_price? There are also supplier pricelists

File: account_invoice_pricelist/view/view.xml [hunk 4]
```diff
@@ -26,9 +26,15 @@
...
             <field name="inherit_id" ref="account.invoice_form"/>
             <field name="arch" type="xml">
+                <field name="state" position="before">
+                    <button type="object" name="button_update_prices_from_pricelist" string="Update prices from pricelist" attrs="{'invisible': ['|', ('pricelist_id', '=', False), ('state', 'not in', ['draft'])]}" />
+                </field>
                 <field name="user_id" position="after">
                     <field name="pricelist_id"/>
                 </field>
+                <field name="invoice_line" position="attributes">
+                    <attribute name="context">{'partner_id': partner_id, 'price_type': context.get('price_type') or False, 'type': type, 'pricelist_id': pricelist_id}</attribute>
```
Comment (pedrobaeza) [inline account_invoice_pricelist/view/view.xml:36]: I prefer that you use here _web_context_tunnel_ to avoid the overwriting of the context key

**Comment by pedrobaeza** [comment]
@hbrunn, are you going to work on this in a moment soon?

**Comment by pedrobaeza** [comment]
I'm seeing that you need to take also into account the change on the quantity, because pricelists can have a different price.

File: account_invoice_pricelist/view/view.xml (hunk 4 above)
Comment (pedrobaeza) [inline account_invoice_pricelist/view/view.xml:36]: Actually, you only need to pass pricelist_id to the context.

**Comment by hbrunn** [comment]
superseded by #215

**Pull request closed without merging**